### ISSUE TITLE
[User Interface: Redesign]: Computational Tool card redesign

### ISSUE THREAD

**Issue by gtsueng** [NIAID-Data-Ecosystem/nde-portal#240]
### Issue Name

Computational Tool card redesign

### Issue Description

We have previously developed a ComputationalTool schema and harvested metadata from a number of software/tool repositories and harmonized the metadata to the schema. During launch, the ComputationalTools were removed from the site altogether as they did not meet standards. WBS discussions following the launch down-prioritized and deferred ComputationalTool-related tasks.

While many ComputationalTool records would be much improved with some standardization and augmentation, these improvements would not mean much if the UI does not display the metadata elements that are useful for evaluating computational tools.

The Ask: We seek approval to start work on redesigning the display of ComputationalTool records in the NIAID Data Ecosystem Discovery Portal.

### Issue Discussion

Revisiting Computational Tools was discussed at the biweekly meeting dated 2024.05.14

### WCAG Compliance Check

- [ ] The ui changes are not expected to significantly affect WCAG compliance (ex- wording changes)
- [X] The ui changes have visual elements which have yet to be tested for WCAG compliance
- [ ] The ui changes have visual elements which have been tested for WCAG compliance
- [ ] The ui changes have passed all checks for WCAG compliance

### Related WBS task

- [ ] Check here if the ui change is centered on the look and feel (https://github.com/NIAID-Data-Ecosystem/nde-roadmap/issues/52)
- [X] Check here if the ui change includes new features (https://github.com/NIAID-Data-Ecosystem/nde-roadmap/issues/53)
- [ ] Check here if the ui change was suggested by the NIAID team (https://github.com/NIAID-Data-Ecosystem/nde-roadmap/issues/2)
- [X] https://github.com/NIAID-Data-Ecosystem/nde-roadmap/issues/29

### For internal use only. Assignee, please select the status of this issue

- [ ] Not yet started
- [ ] In progress
- [ ] Blocked
- [ ] Will not address

### UI change status check list

- [ ] This ui change has yet to be discussed between NIAID, Scripps, Leidos
- [X] This ui change has been discussed/reported between NIAID, Scripps, Leidos
- [x] This ui change has been mocked up
- [x] This ui change has been implemented on Dev
- [x] This ui change has been approved for Staging
- [x] This ui change has been implemented on Staging
- [x] This ui change has been approved for Production
- [x] This ui change has been implemented on Production

**Comment by gtsueng** [comment]
Based on the discussion during the biweekly meeting dated 2024.05.28 we will move forward with re-designing the computational tool card based on the bio.tools resource metadata.

**Comment by gtsueng** [comment]
@leandrocollares regarding this issue, the tool repository with the most organized/complete metadata seems to be bio.tools, so feel free to check out their site for inspiration.

The schema for ComputationalTools can be found here: https://discovery.biothings.io/ns/nde/nde:ComputationalTool

There are a lot of overlapping properties; however, the metadata people might consider filtering on for ComputationalTools may differ greatly from Datasets, so some coverage studies and Lyssna studies may be necessary to prioritize the metadata elements to display and/or include in the filters

**Comment by gtsueng** [comment]
Hi @hartwickma, @lisa-mml, @rshabman, @sudvenk,

Three mock-ups for the computational tool redesign are now available here: https://docs.google.com/presentation/d/1kkVdSEi-loxmc1aRm8vBcAQyy9Dg99Tfb5Zl0LzEN4c/edit#slide=id.g30742cc30d4_0_0

We ran a convenience sampling (no cost) Lyssna study on the three designs, but the results were fairly evenly split. https://docs.google.com/presentation/d/1q016iDAvsakEqzq8Y8xG6DIF6I3yBB0Rcoxr48z0jpk/edit#slide=id.g3040c568d7c_0_15

The Ask: Please let us know which (if any) design NIAID would like to move to staging. If there's no strong preference, or if you'd like us to merge some of the designs, just let us know and we'd be happy to do so.

**Comment by hartwickma** [comment]
Hi @gtsueng, thank you for providing an overview of these design options in our bi-weekly meeting 15Oct2024. The takeaway from our discussion was that Option 1 & 3 had the preferred design features for the Computational Tools card. The key differences between 1 or 3 were that Option 1 included the programming language and workflow is part of the metadata whereas Option 3 has the workflow metadata subset into separate pills. We would like to confirm that these are the differences between design 1 and 3. If so, please proceed with Design 1 to staging. 

NIAID has also requested (as part of the re-deployment) that the Computational Tools indexed in the Ecosystem will be either confirmed as high quality tools  or from a curated list of tools that are essential/relevant to IID. Please provide the procedure and criteria that were used to select the tools that will be indexed in the Portal as Computational Tools.

**Comment by gtsueng** [comment]
Hi @hartwickma, @sudvenk

Confirmed, those are the primary differences between design 1 and 3. We will proceed with moving Design 1 to staging

The selection for moving the bio.tools repository forward to staging is based on the following:
* bio.tools is sponsored by ELIXIR which also sponsors the Bioschemas community. As a result the metadata is highly compatible with the Bioschemas ComputationalTool profile schema. Both the NIAID SysBio and NIAID Data Ecosystem ComputationalTool schemas were derived from the Bioschemas ComputationalTool profile schema.
* bio.tools has [well-documented curation guidelines](https://biotools.readthedocs.io/en/latest/curators_guide.html) so many metadata records indexed in bio.tools have more complete metadata than records indexed in other tool indices
* bio.tools uses EDAM topics ontology for categorizing the domains to which their tools apply. This allows for additional sub-setting of the tools as needed.
* bio.tools has communities of practice (eg. COVID-19 community), which allows another path for additional sub-setting as needed.
* bio.tools curation and the bio.tools registry improvements have been the subject of multiple hackathon projects at BioHackEU, DBLCs Hackathons and more.

**Comment by gtsueng** [comment]
@hartwickma, @rshabman, @lisa-mml, @sudvenk

The re-designed ComputationalTool cards are now on Staging along with the bio.tools repository. As mentioned at the meeting dated 2024.11.12, we are still working on the updating the ComputationalTool Resources page and documentation updates.

Updates to the filters can come after that as that would need to be implemented along with the toggle for routing user searches to either a ComputationalTool or Dataset

**Comment by hartwickma** [comment]
Hi @gtsueng, thank you for this update. It is unclear if there is any action needed on this item from the NIAID end? Please let us know if there are items we need to address otherwise we have changed the status of this to 'Review Provided'.

**Comment by gtsueng** [comment]
Hi @hartwickma, 

The status has been changed to `Review Provided`, we will continue to provide updates via the GH issue, and will change the status of this issue once again, when it is ready to move from Staging to Production

**Comment by gtsueng** [comment]
Hi @hartwickma, @rshabman, @lisa-mml, @sudvenk,

The ComputationalTool-related UI changes are now on Staging **awaiting approval** for Production. This includes updates for:
* the cards (this issue) 
* the [resources page](https://github.com/NIAID-Data-Ecosystem/nde-portal/issues/267)
* the [landing page search options](https://github.com/NIAID-Data-Ecosystem/nde-portal/issues/242)

The following updates cannot be applied to staging (due to the STRAPI pages already being published), so the updates will need to be approved in google docs before moving to production:
* the [documentation](https://github.com/NIAID-Data-Ecosystem/nde-portal/issues/268)

The Ask: We seek approval to move the above to Production

The following changes are under investigation:
* [Facets and filter updates](https://github.com/NIAID-Data-Ecosystem/nde-portal/issues/241)

**Comment by hartwickma** [comment]
Hi @gtsueng once the changes the documentation are finalized please move the computation card redesign to production. 

For future issues with interdependencies please:

1. Avoid adding duplicate notes - it is difficult to determine what to address in order to move the issue forward. Please limit the documentation to what needs to be approved only for that GH issue. If it is too complex for a single GH issue, please provide as an email or an agenda item. 
2. Note that moving forward, NIAID will provide a single review for updates to documentation, text, and aesthetics. Please update NIAID once these changes have been implemented and moved to production.

**Comment by gtsueng** [comment]
@hartwickma @rshabman @lisa-mml @sudvenk

The ComputationalTools are now on Production. I am marking this issue as `pending close out`. 

Regarding multiple issue-- moving forward, I will create 1 primary issue for NIAID to track

**Comment by hartwickma** [comment]
Hi @gtsueng, we would like to revisit part of the discussion from this thread: 

**Oct 16th**
_NIAID has also requested (as part of the re-deployment) that the Computational Tools indexed in the Ecosystem will be either confirmed as high quality tools or from a curated list of tools that are essential/relevant to IID. Please provide the procedure and criteria that were used to select the tools that will be indexed in the Portal as Computational Tools._

We will need to explore technical solutions to incorporate additional computational tools as an individual tool or and/or additional computational tool repositories and ensure that these tools meet community software standards. Can we start a new GitHub issue around technical solutions for adding in additional computational tools and repositories and potential 'Tool Collections' similiar to Program Collections. We would like to discuss at an upcoming meeting.

**Comment by gtsueng** [comment]
Hi @hartwickma, @sudvenk,

A new issue has been generated per this request: https://github.com/NIAID-Data-Ecosystem/niaid-feedback/issues/179
